### PR TITLE
Remove redundant validateOwnedCRDs impl

### PR DESCRIPTION
**Pull request by jchunkins**
**Description of the change:**
- Adjust RegistryBundleValidator to remove the duplicate implementation for validating Owned CRDs that operates on the operator registry bundle type.
- Leave RegistryBundleValidator skeleton in place for future registry bundle validation
- Adjust existing unit test to account for validation removal
- Make ValidateBundleContent use the registry bundle AND api bundle validators
- Fix bug in argument handling for ObjectValidator which would cause validation to never trigger because it was passing an array itself instead of expanding ... to pass to Variadic argument
- enhance test coverage for files that have changed

Signed-off-by: John Hunkins <jhunkins@us.ibm.com>

**Motivation for the change:**

Remove redundant "validate owned CRD" implementation from operator registry while keeping a future point of expansion for registry specific validation. This consolidates the "validate owned CRD" into a single implementation which already exists in the api project.

Also used this as an opportunity to enhance code coverage.

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

Closes https://github.com/operator-framework/operator-registry/issues/640

### CODE REVIEW

File: pkg/lib/bundle/validate.go [hunk 1]
```diff
@@ -99,10 +99,10 @@ func (i imageValidator) ValidateBundleFormat(directory string) error {
 		}
 	}
 
-	if manifestsFound == false {
+	if !manifestsFound {
```
Comment (timflannagan) [inline pkg/lib/bundle/validate.go:102]: :+1:

**Comment by timflannagan** [comment]
I took a quick look at the changes and they seem reasonable but my brain is dead for the day. Going to try and carve out sometime later this week to take another look.

**Comment by jchunkins** [comment]
Thanks @timflannagan

**Comment by awgreene** [comment]
This PR has been open for over a year and seems stale. I am going to close this PR for now, but if the suggested features are still desired, please reopen.

**Pull request closed without merging**